fplc_serial_manager.py

```
@@ -0,0 +1,347 @@
+            self.timestamp = time.time()
+
+    class ErrorHandler:
+        def __init__(self, log_file="fplc_errors.log"):
+            self.log_file = log_file
+            self.error_callbacks = []
+            logging.basicConfig(filename=self.log_file, level=logging.INFO,
+                                format='%(asctime)s - %(levelname)s - %(message)s')
+        def handle_error(self, error: FPLCError):
+            logging.error(f"FPLC Error: {error.message} - {error.severity.value} - {error.details}")
+            for callback in self.error_callbacks:
+                try: callback({'message': error.message, 'severity': error.severity.value, 'details': error.details})
```

Ruff (0.12.2)

Multiple statements on one line (colon)

(E701)